packages/server/src/routes/api/login.ts

```diff
@@ -0,0 +1,94 @@
+import config from '../../config';
+import Router from '../../utils/Router';
+import { redirect, SubPath } from '../../utils/routeUtils';
+import { generateRedirectHtml, identityProvider, serviceProvider } from '../../utils/saml';
+import { AppContext, RouteType, SamlPostResponse } from '../../utils/types';
+import { bodyFields } from '../../utils/requestUtils';
+import { InternalServerError } from '../../utils/errors';
+import { FlowResult } from 'samlify/types/src/flow';
+import { cookieSet } from '../../utils/cookies';
+import defaultView from '../../utils/defaultView';
+
+export const router = new Router(RouteType.Api);
+
+router.public = true;
+
+// Set an error message saying that SAML is disabled.
+// @param ctx The AppContext.
+function samlNotAvailable(ctx: AppContext) {
+	ctx.status = 403;
+	ctx.body = { error: 'This server does not accept SAML authentication.' };
+}
```

Comment:
Please throw an error - see utils/errors.ts. We avoid setting the context from handlers because it's harder to test.